src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -198,8 +198,7 @@ fun <T> CancellableContinuation<T>.cancelByLincheck(promptCancellation: Boolean)
         throw it.cause!! // let's throw the original exception, ignoring the internal coroutines details
     }
     if (!cancelled && promptCancellation) {
-        // TODO we can invoke the method directly if the transformation is disabled
-        getMethod(this, cancelCompletedResultMethod).invoke(this, null, cancellationByLincheckException)
+        context[Job]!!.cancel() // we should always put a job into the context for prompt cancellation
```

Comment:
As far as I now `CancellableContinuationImpl.cancel` does not call `cancelCompletedResult`, so I am not sure that this change is correct...

Reply:
Here we cancel the parent job (not the continuation!) to cancel the resumed but not dispatched continuation -- that is how prompt cancellation works. I had a conversation with Vsevolod Tolstopyatov on this.